// oci-database/lib/model/autonomous-database-backup-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutonomousDatabaseBackupSummary = void 0;
const model = __importStar(require("../model"));
var AutonomousDatabaseBackupSummary;
(function (AutonomousDatabaseBackupSummary) {
    let Type;
    (function (Type) {
        Type["Incremental"] = "INCREMENTAL";
        Type["Full"] = "FULL";
        Type["Longterm"] = "LONGTERM";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = AutonomousDatabaseBackupSummary.Type || (AutonomousDatabaseBackupSummary.Type = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Updating"] = "UPDATING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = AutonomousDatabaseBackupSummary.LifecycleState || (AutonomousDatabaseBackupSummary.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "backupDestinationDetails": obj.backupDestinationDetails
                ? model.BackupDestinationDetails.getJsonObj(obj.backupDestinationDetails)
                : undefined
        });
        return jsonObj;
    }
    AutonomousDatabaseBackupSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "backupDestinationDetails": obj.backupDestinationDetails
                ? model.BackupDestinationDetails.getDeserializedJsonObj(obj.backupDestinationDetails)
                : undefined
        });
        return jsonObj;
    }
    AutonomousDatabaseBackupSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutonomousDatabaseBackupSummary = exports.AutonomousDatabaseBackupSummary || (exports.AutonomousDatabaseBackupSummary = {}));
