// oci-database/lib/model/change-dataguard-role-details.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeDataguardRoleDetails = void 0;
var ChangeDataguardRoleDetails;
(function (ChangeDataguardRoleDetails) {
    let Role;
    (function (Role) {
        Role["Primary"] = "PRIMARY";
        Role["Standby"] = "STANDBY";
        Role["DisabledStandby"] = "DISABLED_STANDBY";
        Role["BackupCopy"] = "BACKUP_COPY";
        Role["SnapshotStandby"] = "SNAPSHOT_STANDBY";
    })(Role = ChangeDataguardRoleDetails.Role || (ChangeDataguardRoleDetails.Role = {}));
    let ConnectionStringsType;
    (function (ConnectionStringsType) {
        ConnectionStringsType["SnapshotServices"] = "SNAPSHOT_SERVICES";
        ConnectionStringsType["PrimaryServices"] = "PRIMARY_SERVICES";
    })(ConnectionStringsType = ChangeDataguardRoleDetails.ConnectionStringsType || (ChangeDataguardRoleDetails.ConnectionStringsType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ChangeDataguardRoleDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ChangeDataguardRoleDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(ChangeDataguardRoleDetails = exports.ChangeDataguardRoleDetails || (exports.ChangeDataguardRoleDetails = {}));
