// oci-database/lib/model/db-backup-config.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbBackupConfig = void 0;
const model = __importStar(require("../model"));
var DbBackupConfig;
(function (DbBackupConfig) {
    let AutoBackupWindow;
    (function (AutoBackupWindow) {
        AutoBackupWindow["SlotOne"] = "SLOT_ONE";
        AutoBackupWindow["SlotTwo"] = "SLOT_TWO";
        AutoBackupWindow["SlotThree"] = "SLOT_THREE";
        AutoBackupWindow["SlotFour"] = "SLOT_FOUR";
        AutoBackupWindow["SlotFive"] = "SLOT_FIVE";
        AutoBackupWindow["SlotSix"] = "SLOT_SIX";
        AutoBackupWindow["SlotSeven"] = "SLOT_SEVEN";
        AutoBackupWindow["SlotEight"] = "SLOT_EIGHT";
        AutoBackupWindow["SlotNine"] = "SLOT_NINE";
        AutoBackupWindow["SlotTen"] = "SLOT_TEN";
        AutoBackupWindow["SlotEleven"] = "SLOT_ELEVEN";
        AutoBackupWindow["SlotTwelve"] = "SLOT_TWELVE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AutoBackupWindow["UnknownValue"] = "UNKNOWN_VALUE";
    })(AutoBackupWindow = DbBackupConfig.AutoBackupWindow || (DbBackupConfig.AutoBackupWindow = {}));
    let AutoFullBackupWindow;
    (function (AutoFullBackupWindow) {
        AutoFullBackupWindow["SlotOne"] = "SLOT_ONE";
        AutoFullBackupWindow["SlotTwo"] = "SLOT_TWO";
        AutoFullBackupWindow["SlotThree"] = "SLOT_THREE";
        AutoFullBackupWindow["SlotFour"] = "SLOT_FOUR";
        AutoFullBackupWindow["SlotFive"] = "SLOT_FIVE";
        AutoFullBackupWindow["SlotSix"] = "SLOT_SIX";
        AutoFullBackupWindow["SlotSeven"] = "SLOT_SEVEN";
        AutoFullBackupWindow["SlotEight"] = "SLOT_EIGHT";
        AutoFullBackupWindow["SlotNine"] = "SLOT_NINE";
        AutoFullBackupWindow["SlotTen"] = "SLOT_TEN";
        AutoFullBackupWindow["SlotEleven"] = "SLOT_ELEVEN";
        AutoFullBackupWindow["SlotTwelve"] = "SLOT_TWELVE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AutoFullBackupWindow["UnknownValue"] = "UNKNOWN_VALUE";
    })(AutoFullBackupWindow = DbBackupConfig.AutoFullBackupWindow || (DbBackupConfig.AutoFullBackupWindow = {}));
    let AutoFullBackupDay;
    (function (AutoFullBackupDay) {
        AutoFullBackupDay["Sunday"] = "SUNDAY";
        AutoFullBackupDay["Monday"] = "MONDAY";
        AutoFullBackupDay["Tuesday"] = "TUESDAY";
        AutoFullBackupDay["Wednesday"] = "WEDNESDAY";
        AutoFullBackupDay["Thursday"] = "THURSDAY";
        AutoFullBackupDay["Friday"] = "FRIDAY";
        AutoFullBackupDay["Saturday"] = "SATURDAY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AutoFullBackupDay["UnknownValue"] = "UNKNOWN_VALUE";
    })(AutoFullBackupDay = DbBackupConfig.AutoFullBackupDay || (DbBackupConfig.AutoFullBackupDay = {}));
    let BackupDeletionPolicy;
    (function (BackupDeletionPolicy) {
        BackupDeletionPolicy["DeleteImmediately"] = "DELETE_IMMEDIATELY";
        BackupDeletionPolicy["DeleteAfterRetentionPeriod"] = "DELETE_AFTER_RETENTION_PERIOD";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        BackupDeletionPolicy["UnknownValue"] = "UNKNOWN_VALUE";
    })(BackupDeletionPolicy = DbBackupConfig.BackupDeletionPolicy || (DbBackupConfig.BackupDeletionPolicy = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "backupDestinationDetails": obj.backupDestinationDetails
                ? obj.backupDestinationDetails.map(item => {
                    return model.BackupDestinationDetails.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DbBackupConfig.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "backupDestinationDetails": obj.backupDestinationDetails
                ? obj.backupDestinationDetails.map(item => {
                    return model.BackupDestinationDetails.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    DbBackupConfig.getDeserializedJsonObj = getDeserializedJsonObj;
})(DbBackupConfig = exports.DbBackupConfig || (exports.DbBackupConfig = {}));
