// oci-database/lib/model/infrastructure-target-version-summary.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfrastructureTargetVersionSummary = void 0;
var InfrastructureTargetVersionSummary;
(function (InfrastructureTargetVersionSummary) {
    let TargetResourceType;
    (function (TargetResourceType) {
        TargetResourceType["ExadataDbSystem"] = "EXADATA_DB_SYSTEM";
        TargetResourceType["CloudExadataInfrastructure"] = "CLOUD_EXADATA_INFRASTRUCTURE";
        TargetResourceType["ExaccInfrastructure"] = "EXACC_INFRASTRUCTURE";
    })(TargetResourceType = InfrastructureTargetVersionSummary.TargetResourceType || (InfrastructureTargetVersionSummary.TargetResourceType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    InfrastructureTargetVersionSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    InfrastructureTargetVersionSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(InfrastructureTargetVersionSummary = exports.InfrastructureTargetVersionSummary || (exports.InfrastructureTargetVersionSummary = {}));
