// oci-database/lib/model/long-term-back-up-schedule-details.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LongTermBackUpScheduleDetails = void 0;
var LongTermBackUpScheduleDetails;
(function (LongTermBackUpScheduleDetails) {
    let RepeatCadence;
    (function (RepeatCadence) {
        RepeatCadence["OneTime"] = "ONE_TIME";
        RepeatCadence["Weekly"] = "WEEKLY";
        RepeatCadence["Monthly"] = "MONTHLY";
        RepeatCadence["Yearly"] = "YEARLY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RepeatCadence["UnknownValue"] = "UNKNOWN_VALUE";
    })(RepeatCadence = LongTermBackUpScheduleDetails.RepeatCadence || (LongTermBackUpScheduleDetails.RepeatCadence = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LongTermBackUpScheduleDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    LongTermBackUpScheduleDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(LongTermBackUpScheduleDetails = exports.LongTermBackUpScheduleDetails || (exports.LongTermBackUpScheduleDetails = {}));
