// oci-database/lib/model/saas-admin-user-configuration.js
"use strict";
/**
 * Database Service API
 * The API for the Database Service. Use this API to manage resources such as databases and DB Systems. For more information, see [Overview of the Database Service](/iaas/Content/Database/Concepts/databaseoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: sic_dbaas_cp_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SaasAdminUserConfiguration = void 0;
var SaasAdminUserConfiguration;
(function (SaasAdminUserConfiguration) {
    let AccessType;
    (function (AccessType) {
        AccessType["ReadOnly"] = "READ_ONLY";
        AccessType["ReadWrite"] = "READ_WRITE";
        AccessType["Admin"] = "ADMIN";
    })(AccessType = SaasAdminUserConfiguration.AccessType || (SaasAdminUserConfiguration.AccessType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SaasAdminUserConfiguration.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    SaasAdminUserConfiguration.getDeserializedJsonObj = getDeserializedJsonObj;
})(SaasAdminUserConfiguration = exports.SaasAdminUserConfiguration || (exports.SaasAdminUserConfiguration = {}));
