// oci-core/lib/model/address-type.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](/iaas/Content/Network/Concepts/overview.htm),
[Compute](/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](/iaas/Content/Block/Concepts/overview.htm) services.
The required permissions are documented in the
[Details for the Core Services](/iaas/Content/Identity/Reference/corepolicyreference.htm) article.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressType = void 0;
/**
 * Address type of the CIDR/IP within a VCN or subnet
 **/
var AddressType;
(function (AddressType) {
    AddressType["PrivateIPv4"] = "Private_IPv4";
    AddressType["OracleAllocatedPublicIPv4"] = "Oracle_Allocated_Public_IPv4";
    AddressType["ByoipIPv4"] = "BYOIP_IPv4";
    AddressType["UlaIPv6"] = "ULA_IPv6";
    AddressType["OracleAllocatedGuaIPv6"] = "Oracle_Allocated_GUA_IPv6";
    AddressType["ByoipIPv6"] = "BYOIP_IPv6";
})(AddressType = exports.AddressType || (exports.AddressType = {}));
(function (AddressType) {
    function getJsonObj(obj) {
        return obj;
    }
    AddressType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AddressType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AddressType = exports.AddressType || (exports.AddressType = {}));
