// oci-database/lib/request/list-autonomous-container-database-versions-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAutonomousContainerDatabaseVersionsRequest = void 0;
var ListAutonomousContainerDatabaseVersionsRequest;
(function (ListAutonomousContainerDatabaseVersionsRequest) {
    let ServiceComponent;
    (function (ServiceComponent) {
        ServiceComponent["Adbd"] = "ADBD";
        ServiceComponent["Exacc"] = "EXACC";
    })(ServiceComponent = ListAutonomousContainerDatabaseVersionsRequest.ServiceComponent || (ListAutonomousContainerDatabaseVersionsRequest.ServiceComponent = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAutonomousContainerDatabaseVersionsRequest.SortOrder || (ListAutonomousContainerDatabaseVersionsRequest.SortOrder = {}));
})(ListAutonomousContainerDatabaseVersionsRequest = exports.ListAutonomousContainerDatabaseVersionsRequest || (exports.ListAutonomousContainerDatabaseVersionsRequest = {}));
