// oci-database/lib/request/list-maintenance-run-history-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMaintenanceRunHistoryRequest = void 0;
var ListMaintenanceRunHistoryRequest;
(function (ListMaintenanceRunHistoryRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeScheduled"] = "TIME_SCHEDULED";
        SortBy["TimeEnded"] = "TIME_ENDED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListMaintenanceRunHistoryRequest.SortBy || (ListMaintenanceRunHistoryRequest.SortBy = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMaintenanceRunHistoryRequest.SortOrder || (ListMaintenanceRunHistoryRequest.SortOrder = {}));
})(ListMaintenanceRunHistoryRequest = exports.ListMaintenanceRunHistoryRequest || (exports.ListMaintenanceRunHistoryRequest = {}));
